### PR TITLE
[SPARK-39229][SQL] Separate query contexts from error-classes.json

### PR DESCRIPTION
**Pull request by gengliangwang**
### What changes were proposed in this pull request?

Separate query contexts for runtime errors from error-classes.json.

### Why are the changes needed?

The message is JSON should only contain parameters explicitly thrown. It is more elegant to separate query contexts from error-classes.json.

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

Existing UT

### CODE REVIEW

**Review by cloud-fan — APPROVED** [review]

File: core/src/main/scala/org/apache/spark/ErrorInfo.scala [hunk 1]
```diff
@@ -82,11 +85,11 @@ private[spark] object SparkThrowableHelper {
       val subMessageParameters = messageParameters.tail
       "[" + errorClass + "." + subErrorClass + "] " + String.format((errorInfo.messageFormat +
         errorSubInfo.messageFormat).replaceAll("<[a-zA-Z0-9_-]+>", "%s"),
-        subMessageParameters: _*)
+        subMessageParameters: _*) + queryContext
```
Comment (srielau) [inline core/src/main/scala/org/apache/spark/ErrorInfo.scala:88]: I know it's causing diffs, but we really should have a space.

```suggestion
        subMessageParameters: _*) + " " + queryContext
```

File: core/src/main/scala/org/apache/spark/ErrorInfo.scala [hunk 2]
```diff
@@ -82,11 +85,11 @@ private[spark] object SparkThrowableHelper {
...
     } else {
       "[" + errorClass + "] " + String.format(
         errorInfo.messageFormat.replaceAll("<[a-zA-Z0-9_-]+>", "%s"),
-        messageParameters: _*)
+        messageParameters: _*) + queryContext
```
Comment (srielau) [inline core/src/main/scala/org/apache/spark/ErrorInfo.scala:92]:
```suggestion
        messageParameters: _*) + " " + queryContext
```

**Review by srielau — APPROVED** [review]

File: core/src/main/scala/org/apache/spark/ErrorInfo.scala (hunk 1 above)
Comment (gengliangwang) [reply core/src/main/scala/org/apache/spark/ErrorInfo.scala:88]: Actually, there is a "\n" at the beginning of `queryContext`.
Probably I should move it from `queryContext` to the method here.

Comment (gengliangwang) [reply core/src/main/scala/org/apache/spark/ErrorInfo.scala:88]: I tried moving the "\n" out of the query context. It requires updates on the starting and ending position of "^^^^^^" too. I will do it in another PR to make this one clean.

**Review by MaxGekk — APPROVED** [review]
+1, LGTM. Merging to master.
Thank you, @gengliangwang and @cloud-fan @srielau for review.

**Comment by MaxGekk** [comment]
@gengliangwang The changes cause conflicts in branch-3.3. Please, open a separate PR to backport this (if it is needed)

**Comment by gengliangwang** [comment]
@MaxGekk I have opened a backport in https://github.com/apache/spark/pull/36607

File: core/src/main/scala/org/apache/spark/ErrorInfo.scala (hunk 1 above)
Comment (gengliangwang) [reply core/src/main/scala/org/apache/spark/ErrorInfo.scala:88]: Opened https://github.com/apache/spark/pull/36612 for this.

**Pull request closed without merging**